x-pack/plugins/cloud/server/plugin.ts

```diff
@@ -201,6 +202,24 @@ export class CloudPlugin implements Plugin<CloudSetup, CloudStart> {
       orchestratorTarget,
     });
 
+    core.http.registerOnPreAuth((request, response, toolkit) => {
+      const queryOnboardingToken = request.url.searchParams.get('onboarding_token');
+      if (queryOnboardingToken) {
```

Comment:
if we are in this condition, It might be good to remove the query string after all by doing something like that

```
response.redirected({ headers: { location: newDestination } });
```